server/api/cas/group/new.post.ts

```diff
@@ -0,0 +1,56 @@
+import { PrismaClient } from '@prisma/client'
+import * as z from 'zod'
+
+const prisma = new PrismaClient()
+
+const requestSchema = z.object({
+  clubId: z.string(),
+  wechatGroupUrl: z.string().startsWith('https://weixin.qq.com/g/', { message: 'WeChat Group URL required' }),
+  wechatGroupExpiration: z.string().datetime(),
+})
+
+export default eventHandler(async (event) => {
+  const { auth } = event.context
+
+  if (!auth.userId) {
+    setResponseStatus(event, 403)
+    return
+  }
+
+  const tsimsStudentId = await prisma.user.findUnique({
+    where: {
+      id: auth.userId,
+    },
+  }).then(user => user?.tsimsStudentId)
+
+  const requestBody = await readValidatedBody(event, body => requestSchema.parse(body))
+
+  const isPresidentOrVicePresident = await prisma.club.findFirst({
+    where: {
+      id: Number(requestBody.clubId),
+      OR: [
+        {
+          presidentByTsimsStudentId: tsimsStudentId,
+        },
+        {
+          vicesByTsimsStudentId: {
+            has: tsimsStudentId,
+          },
+        },
+      ],
+    },
+  }).then(club => club !== null)
```

Comment:
Just find a club which has the given id and `presidentByTsimsStudentId === tsimsStudentId`.

Reply:
If undefined, return 403.